<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>COMMOT &mdash; commot 0.0.3 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/sphinx_rtd_size.css" type="text/css" />
      <link rel="stylesheet" href="_static/sg_gallery.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
        <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Installation" href="installation.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
            <a href="#" class="icon icon-home"> commot
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <ul>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorials.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="api/index.html">API</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">commot</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
      <li>COMMOT</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  
<style>
/* CSS overrides for sphinx_rtd_theme */

/* 24px margin */
.nbinput.nblast.container,
.nboutput.nblast.container {
    margin-bottom: 19px;  /* padding has already 5px */
}

/* ... except between code cells! */
.nblast.container + .nbinput.container {
    margin-top: -19px;
}

.admonition > p:before {
    margin-right: 4px;  /* make room for the exclamation icon */
}

/* Fix math alignment, see https://github.com/rtfd/sphinx_rtd_theme/pull/686 */
.math {
    text-align: unset;
}
</style>
<section id="commot">
<h1>COMMOT<a class="headerlink" href="#commot" title="Permalink to this heading"></a></h1>
<p>Screening cell-cell communication in spatial transcriptomics via collective optimal transport</p>
<a class="reference external image-reference" href="https://github.com/zcang/COMMOT/actions/workflows/python-package.yml/badge.svg"><img alt="pytest" src="https://github.com/zcang/COMMOT/actions/workflows/python-package.yml/badge.svg" /></a>
<section id="installation">
<h2>Installation<a class="headerlink" href="#installation" title="Permalink to this heading"></a></h2>
<p>Install from PyPI by <code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">commot</span></code> or install from source by cd to this directory and <code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">.</span></code></p>
<p>[Optional] Install <a class="reference external" href="https://github.com/statOmics/tradeSeq">tradeSeq</a> in R to analyze the CCC differentially expressed genes.  <span class="raw-html-md"><br /></span>
Currently, tradeSeq version 1.0.1 with R version 3.6.3 has been tested to work.  <span class="raw-html-md"><br /></span>
In order for the R-python interface to work properly, rpy2==3.4.2 and anndata2ri==1.0.6 should be installed.
To use this feature, install from PyPI by <code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">commot[tradeSeq]</span></code> or from source by <code class="docutils literal notranslate"><span class="pre">pip</span> <span class="pre">install</span> <span class="pre">.[tradeSeq]</span></code>.</p>
</section>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Permalink to this heading"></a></h2>
<p><strong>Basic usage</strong></p>
<p><em>Import packages</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">commot</span> <span class="k">as</span> <span class="nn">ct</span>
<span class="kn">import</span> <span class="nn">scanpy</span> <span class="k">as</span> <span class="nn">sc</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
</pre></div>
</div>
<p><em>Load a spatial dataset</em>  <span class="raw-html-md"><br /></span>
(e.g., a Visium dataset)</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">adata</span> <span class="o">=</span> <span class="n">sc</span><span class="o">.</span><span class="n">datasets</span><span class="o">.</span><span class="n">visium_sge</span><span class="p">(</span><span class="n">sample_id</span><span class="o">=</span><span class="s1">&#39;V1_Mouse_Brain_Sagittal_Posterior&#39;</span><span class="p">)</span>
<span class="n">adata</span><span class="o">.</span><span class="n">var_names_make_unique</span><span class="p">()</span>
</pre></div>
</div>
<p><em>Basic processing</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">normalize_total</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span> <span class="n">inplace</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">sc</span><span class="o">.</span><span class="n">pp</span><span class="o">.</span><span class="n">log1p</span><span class="p">(</span><span class="n">adata</span><span class="p">)</span>
</pre></div>
</div>
<p><em>Specify ligand-receptor pairs</em></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">LR</span><span class="o">=</span><span class="n">np</span><span class="o">.</span><span class="n">array</span><span class="p">([[</span><span class="s1">&#39;Tgfb1&#39;</span><span class="p">,</span> <span class="s1">&#39;Tgfbr1_Tgfbr2&#39;</span><span class="p">,</span> <span class="s1">&#39;Tgfb_pathway&#39;</span><span class="p">],[</span><span class="s1">&#39;Tgfb2&#39;</span><span class="p">,</span> <span class="s1">&#39;Tgfbr1_Tgfbr2&#39;</span><span class="p">,</span> <span class="s1">&#39;Tgfb_pathway&#39;</span><span class="p">],[</span><span class="s1">&#39;Tgfb3&#39;</span><span class="p">,</span> <span class="s1">&#39;Tgfbr1_Tgfbr2&#39;</span><span class="p">,</span> <span class="s1">&#39;Tgfb_pathway&#39;</span><span class="p">]],</span><span class="n">dtype</span><span class="o">=</span><span class="nb">str</span><span class="p">)</span>
<span class="n">df_ligrec</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">LR</span><span class="p">)</span>
</pre></div>
</div>
<p>(or use pairs from a ligand-receptor database <code class="docutils literal notranslate"><span class="pre">df_ligrec=ct.pp.ligand_receptor_database(database='CellChat',</span> <span class="pre">species='mouse')</span></code>.)</p>
<p><em>Construct CCC networks</em>  <span class="raw-html-md"><br /></span>
Use collective optimal transport to construct CCC networks for the ligand-receptor pairs with a spatial distance constraint of 200 (coupling between cells with distance greater than 200 is prohibited). For example, the spot-by-spot matrix for the pair Tgfb1 (ligand) and Tgfbr1_Tgfbr2 (receptor)is stored in <code class="docutils literal notranslate"><span class="pre">adata.obsp['commot-user_database-Tgfb1-Tgfbr1_Tgfbr2']</span></code>. The total sent or received signal for each pair is stored in <code class="docutils literal notranslate"><span class="pre">adata.obsm['commot-user_database-sum-sender']</span></code> and <code class="docutils literal notranslate"><span class="pre">adata.obsm['commot-user_database-sum-receiver']</span></code>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ct</span><span class="o">.</span><span class="n">tl</span><span class="o">.</span><span class="n">spatial_communication</span><span class="p">(</span><span class="n">adata</span><span class="p">,</span>
    <span class="n">database_name</span><span class="o">=</span><span class="s1">&#39;user_database&#39;</span><span class="p">,</span> <span class="n">df_ligrec</span><span class="o">=</span><span class="n">df_ligrec</span><span class="p">,</span> <span class="n">dis_thr</span><span class="o">=</span><span class="mi">200</span><span class="p">,</span> <span class="n">heteromeric</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
</pre></div>
</div>
<p><strong>Documentation</strong></p>
<p>See the documentation at <a class="reference external" href="https://commot.readthedocs.io/en/latest/index.html">https://commot.readthedocs.io/en/latest/index.html</a> for all the APIs to perform visualization and analyses such as visualizing spatial signaling direction and identifying CCC differentially expressed genes.</p>
<p><strong>Reference</strong></p>
<p>Cang, Zixuan, Yanxiang Zhao, Axel A. Almet, Adam Stabell, Raul Ramos, Maksim Plikus, Scott X. Atwood, and Qing Nie. “Screening cell-cell communication in spatial transcriptomics via collective optimal transport.” bioRxiv (2022): 505185</p>
<div class="toctree-wrapper compound">
</div>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="installation.html" class="btn btn-neutral float-right" title="Installation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2021, Zixuan Cang.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>